### PR TITLE
Added viewer and sample

### PR DESCRIPTION
**Pull request by odeheurles**
Created a sample app which uses the engine and log all transitions to a file

image

The viewer can watch this file and will list all transitions

Here I modified the property C in the sample app and the viewer shows the last transition
image

Here we modify A, but an exception occurs when evaluating Add(A, B) and we see in red the nodes which can't be calculated due to the exception.

image

### CODE REVIEW

**Comment by JakeGinnivan** [comment]
This looks awesome!

File: src/ReactGraphViewer/App.config [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+    <startup> 
+        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5" />
+    </startup>
+  <appSettings>
+    <add key="logDirectory" value="C:\Users\olivier.deheurles\_code\ReactGraph\src\SampleApp\bin\Debug"/>
```
Comment (JakeGinnivan) [inline src/ReactGraphViewer/App.config:7]: Should be relative path?

File: src/ReactGraph.Tests/VisualisationTests.AddMetadataToNodesTest.approved.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@
      C [label="c.Value", fillcolor=".7 .3 .5", shape="box", style="filled,rounded"];
      __2 [label="b.Value", shape="box", style="filled,rounded"];
 
-     __1 -> Addition;
-     Addition -> C;
-     __2 -> Addition;
+    __1 -> Addition;
```
Comment (JakeGinnivan) [inline src/ReactGraph.Tests/VisualisationTests.AddMetadataToNodesTest.approved.txt:7]: Not sure if this is intentional, but the indent is now smaller by 1 space

File: src/ReactGraphViewer/App.config (hunk 1 above)
Comment (odeheurles) [inline src/ReactGraphViewer/App.config:7]: Yep I'll fix that.

Olivier

**Pull request merged**